### PR TITLE
Feat/update to vue3

### PR DESCRIPTION
**Pull request by diegopf**
# Pull request template
 
Describe the purpose of the change, the specific changes done in detail, and the issue you have fixed.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [ ] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](https://github.com/empathyco/x/blob/main/.github/contributing/tests.md): 

## Checklist:

- [ ] My code follows the **[style guidelines](https://github.com/empathyco/x/blob/main/.github/CONTRIBUTING.md#style-guides)** of this project.
- [ ] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: src/components/main.vue [hunk 1]
```diff
@@ -44,7 +41,7 @@
       const showRecommendations: ComputedRef<boolean> = computed(
         () => $x.noResults && !$x.partialResults.length && !$x.semanticQueries.length
       );
-      return { hasSearched, semanticsLocation, showRecommendations };
+      return { hasSearched, semanticsLocation, showRecommendations, x: use$x() };
```
Comment (victorcg88) [inline src/components/main.vue:44]:
```suggestion
      return { hasSearched, semanticsLocation, showRecommendations, x: $x };
```

Although I have mixed feelings.

File: src/components/mobile/mobile-close-aside.vue [hunk 2]
```diff
@@ -11,17 +11,22 @@
...
   import { ClearFilters } from '@empathyco/x-components/facets';
   import { defineComponent } from 'vue';
 
   export default defineComponent({
-    components: { BaseIdModalClose, ClearFilters }
+    components: { BaseIdModalClose, ClearFilters },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/mobile/mobile-close-aside.vue:29]:
```suggestion
      return { x: use$x() };
```

File: src/components/mobile/mobile-open-aside.vue [hunk 3]
```diff
@@ -7,20 +7,25 @@
...
+  import { BaseIdModalOpen, FiltersIcon, use$x } from '@empathyco/x-components';
   import { defineComponent } from 'vue';
 
   export default defineComponent({
-    components: { BaseIdModalOpen, FiltersIcon }
+    components: { BaseIdModalOpen, FiltersIcon },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/mobile/mobile-open-aside.vue:28]:
```suggestion
      return { x: use$x() };
```

File: src/components/mobile/mobile-toolbar.vue [hunk 4]
```diff
@@ -1,27 +1,33 @@
...
   import ColumnPicker from '../column-picker.vue';
 
   export default defineComponent({
     components: {
       ColumnPicker
+    },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/mobile/mobile-toolbar.vue:30]:
```suggestion
      return { x: use$x() };
```

File: src/components/predictive-layer/predictive-layer.vue [hunk 5]
```diff
@@ -71,7 +71,7 @@
                     <template #remove-button-content>
                       <span
                         :aria-label="
-                          $t('historyQueries.removeLabel', { suggestion: suggestion.query })
+                          ('historyQueries.removeLabel', { suggestion: suggestion.query })
```
Comment (victorcg88) [inline src/components/predictive-layer/predictive-layer.vue:74]: Is this change intentional or accidental?

File: src/components/predictive-layer/predictive-layer.vue [hunk 6]
```diff
@@ -95,23 +95,23 @@
...
             <BaseIdModalOpen
-              v-if="isDesktopOrGreater && !$x.query.searchBox"
+              v-if="isDesktopOrGreater && !x.query.searchBox"
               modalId="my-history-aside"
               class="x-button-neutral x-button-sm x-button-tight x-self-start"
               data-test="my-history-button"
             >
               <SettingsIcon />
-              {{ $t('myHistory.openButton') }}
+              {{ 'myHistory.openButton' }}
```
Comment (victorcg88) [inline src/components/predictive-layer/predictive-layer.vue:104]: Same as above: Is this change intentional or accidental?

File: src/components/predictive-layer/sliding-recommendations.vue [hunk 7]
```diff
@@ -54,6 +54,11 @@
       Recommendations,
       Result,
       SlidingPanel
+    },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/predictive-layer/sliding-recommendations.vue:61]:
```suggestion
      return { x: use$x() };
```

File: src/components/search/desktop-aside.vue [hunk 8]
```diff
@@ -44,6 +44,11 @@
       BaseIdModalClose,
       CrossIcon,
       ClearFilters
+    },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/search/desktop-aside.vue:51]:
```suggestion
      return { x: use$x() };
```

File: src/components/search/facets/facet-selected-filters.vue [hunk 9]
```diff
@@ -11,13 +11,19 @@
...
     props: {
       selectedFilters: {
         type: Array as PropType<FacetFilter[]>,
         required: true
       }
+    },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/search/facets/facet-selected-filters.vue:26]:
```suggestion
      return { x: use$x() };
```

File: src/components/search/mobile-aside.vue [hunk 10]
```diff
@@ -29,6 +29,11 @@
       MobileCloseAside,
       SelectedFilters,
       Sort
+    },
+    setup() {
+      return {
+        x: use$x()
+      };
```
Comment (victorcg88) [inline src/components/search/mobile-aside.vue:36]:
```suggestion
      return { x: use$x() };
```

File: src/shims-vue.d.ts [hunk 11]
```diff
@@ -1,4 +1,5 @@
 declare module '*.vue' {
-  import Vue from 'vue';
-  export default Vue;
+  import { DefineComponent } from 'vue';
+  const component: DefineComponent<{}, {}, any>;
+  export default component;
```
Comment (victorcg88) [inline src/shims-vue.d.ts:4]:
```suggestion
  const component: DefineComponent;
  export default component;
```

**Review by joseacabaneros — DISMISSED** [review]

**Review by victorcg88 — APPROVED** [review]

**Review by victorcg88 — DISMISSED** [review]

**Review by victorcg88 — APPROVED** [review]

**Review by joseacabaneros — APPROVED** [review]
Good job with the e2e @victorcg88! Nice not to see any webpack dependency finally ✌🏾
